Dockerfile-tests

```diff
@@ -0,0 +1,46 @@
+# We build an image with bee-queue's node_modules and all the source
+# files.  We jump through some Docker FROM hoops to avoid having the
+# bloat of the g++ compiler and other tools in the final image.
+
+FROM node:lts-alpine as beequeue-base
+
+# Info, and add bash, which we need in the final image
+RUN true \
+  && echo -n "node " && node --version \
+  && echo -n "npm " && npm --version \
+  && echo -n "apk " && apk --version \
+  && apk update \
+  && apk add --no-cache bash \
+  && rm -r /usr/share/man \
+  && true
+
+FROM beequeue-base as beequeue-node_modules
+
+# Ensure that tools for npm are available (for node-gyp in particular)
+# to build node_modules
+RUN true \
+  && apk add --no-cache python make g++ \
+  && true
+
+WORKDIR /bee-queue
```

Comment:
Maybe this is a convention that I'm not familiar with, but it strikes me as really weird to introduce new root directories. Would we put this somewhere else like `/opt` or something?

Reply:
Or something... ;-)   Seriously, I work with literally dozens of different Docker containers that have a complex, Docker-based CICD system.  Each container in this universe typically does just one thing (Express server, HTML->PDF renderer, bee-queue workers).  Long ago we tried following file-system guidelines, but it just meant that bits and pieces of the one thing were spread out, and were hard to find, and because of the "or something" factor, not predictable across developers.  The pattern that has emerged on the projects I work on on is to put everything in a top-level directory.  That way, when you have to connect to the container and poke around (via some ssh chain), everything that matters is in that one place.

Reply:
:shrug: works for me!